api-gateway/src/main/scala/airavata/github/captivate/apigateway/ApigatewayRoutes.scala

```diff
@@ -35,14 +35,19 @@ object ApigatewayRoutes {
     }
   }
 
-  def dataRetrievalRoute[F[_]: Sync](): HttpRoutes[F] = {
+  def dataRetrievalRoute[F[_]: Sync](producer: Option[KafkaProducer[String, String]]): HttpRoutes[F] = {
     val dsl = new Http4sDsl[F]{}
     import dsl._
     HttpRoutes.of[F] {
-      case GET -> Root / "data-retrieval"  =>
+      case req@ POST -> Root / "data-retrieval"  => req.decode[String] { data =>
         for {
           resp <- Ok("Success")
```

Comment:
It might be basic but i'm confused we are also validating/handling negative cases?

Reply:
Not yet . We assume only good data for now. validations will come later

Reply:
Okay, thanks